LingoClash/LingoClash/ViewModels/Auth/SignUpViewModel.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  SignUpViewModel.swift
+//  LingoClash
+//
+//  Created by Kyle キラ on 14/3/22.
+//
+
+import Foundation
+import FirebaseAuth
+import FirebaseFirestore
+
+final class SignUpViewModel {
+    
+    @Published var error: String?
+    
+    func signUp(firstName: String, lastName: String, email: String, password: String) {
+        
+        let fields = [
+            "firstName": firstName,
+            "lastName": lastName,
+            "email": email,
+            "password": password
+        ]
+        
+        // Validate fields
+        let error = validateFields(fields)
```

Comment:
I think it's better if you just modify the error message directly and return a boolean true/false here instead.
Something like 
```
if !validateFields(fields) {
  return;
}
```
then just modify `self.error = "Please fill in all fields."` etc directly. It feels weird that a method `validateFields` returns a nil if okay and an error message if not, just my 2 cents

Reply:
Sure, this works as well.